src/ActionList/Heading.tsx

```diff
@@ -0,0 +1,42 @@
+import React, {forwardRef} from 'react'
+import {BetterSystemStyleObject, SxProp, merge} from '../sx'
+import {defaultSxProp} from '../utils/defaultSxProp'
+import {useRefObjectAsForwardedRef} from '../hooks'
+import {ForwardRefComponent as PolymorphicForwardRefComponent} from '../utils/polymorphic'
+import {default as HeadingComponent} from '../Heading'
+import {ListContext} from './List'
+
+export type ActionListHeadingProps = {
+  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'
+} & SxProp
+
+export const Heading = forwardRef(({as: Component = 'h3', children, sx = defaultSxProp, ...props}, forwardedRef) => {
+  const innerRef = React.useRef<HTMLHeadingElement>(null)
+  useRefObjectAsForwardedRef(forwardedRef, innerRef)
+
+  const {headingId: headingId} = React.useContext(ListContext)
+
+  const styles = {
```

Comment:
I copied the default group heading styles but this is probably not true? Is there a place I can refer to for the general actionlist heading styles?

Reply:
>Note: there is a full variant in ActionList with different padding that needs to be considered here

Great point I added this!

>I was expected [Figma: primer web](https://www.figma.com/file/GCvY3Qv8czRgZgvl1dG6lp/Primer-Web?type=design&node-id=6-24956&mode=design&t=295TfMTZyn1RluMk-0) to have the heading but it doens't 🤔

Yeah! I chatted with @langermank and she said she can help us with the design questions 🙌🏻 I'll collect them in one comment in this PR to make it easier.

Reply:
@langermank Here are the questions:

1) I copied the list heading styles from PVC but wanted to make sure if you have any feedback
	- inset variant (default):  [story](https://primer-c4f9c0d65c-13348165.drafts.github.io/storybook/?path=/story/components-actionlist-features--with-heading)
	- full variant: [story](https://primer-c4f9c0d65c-13348165.drafts.github.io/storybook/?path=/story/components-actionlist-features--with-heading-in-full-variant)
2) How are the heading styles look like when it is used in a menu? Do you have any feedback for this case? [story](https://primer-c4f9c0d65c-13348165.drafts.github.io/storybook/?path=/story/components-actionlist-features--with-heading-in-menu)
3) We have [subtle](https://primer-c4f9c0d65c-13348165.drafts.github.io/storybook/?path=/story/components-actionlist-features--group-with-subtle-title) and [filled](https://primer-c4f9c0d65c-13348165.drafts.github.io/storybook/?path=/story/components-actionlist-features--group-with-filled-title) variants for group headings. Do we want the same thing for the list headings? PVC doesn't seem to have them.

Thank you! 🙌🏻